src/org/labkey/test/tests/SampleTypeLinkToStudyTest.java

```diff
@@ -496,17 +507,53 @@ public void testLinkedColumnNotDisplayedCase()
         customizeView.saveCustomView();
     }
 
+    @Test
+    public void testManualDatasetCategoryLink()
+    {
+        String categoryName = "CAT1";
+        createDatasetCategory(VISIT_BASED_STUDY, categoryName);
+        goToProjectHome();
+
+        log("Linking the sample type to preexisting dataset category");
+        linkToStudy(VISIT_BASED_STUDY, SAMPLE_TYPE1, 1, categoryName);
+
+        log("Linking the sample type to new dataset category");
+        goToProjectHome();
+        linkToStudy(DATE_BASED_STUDY, SAMPLE_TYPE2, 1, "CAT2");
+
+        checker().verifyEquals("Incorrect category for the dataset(Category exists case)", categoryName,
+                getCategory(VISIT_BASED_STUDY, SAMPLE_TYPE1));
+        checker().verifyEquals("Incorrect category for the dataset(New category case)", "CAT2",
+                getCategory(DATE_BASED_STUDY, SAMPLE_TYPE2));
+    }
```

Comment:
For consistency here I'd either in-line the `"CAT1"` variable or pull `"CAT2"` into a `String categoryName2 = "CAT2";` variable. Just so the calls to verifyEquals() are parallel and a little more readable.